### PR TITLE
MCS-1501 Update requirements

### PR DESCRIPTION
**Pull request by ThomasSchellenbergNextCentury**
- Updated versions in the requirements file (as much as possible).
- Added github workflows to automatically test python 3.8, 3.9, and 3.10 on both linux and mac.

### CODE REVIEW

File: .github/workflows/ci_3_10.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Install Python 3.10
```
Comment (ThomasSchellenbergNextCentury) [inline .github/workflows/ci_3_10.yaml:19]: I added new github workflows for testing python 3.8, 3.9, and 3.10.

Comment (ThomasSchellenbergNextCentury) [inline .github/workflows/ci_3_10.yaml:19]: I added three new github workflows for automatically running tests on python 3.8, 3.9, and 3.10 (both mac and linux). Previously we just tested python 3.7

File: integration_tests/additional_integration_tests.py [hunk 2]
```diff
@@ -72,7 +72,7 @@ def run_depth_and_segmentation_test(controller, metadata_tier):
         for col_idx, pixel_col in enumerate(pixel_row):
 
             # Need to account for hardware differences
-            if(not math.isclose(
+            if (not math.isclose(
```
Comment (ThomasSchellenbergNextCentury) [inline integration_tests/additional_integration_tests.py:75]: One of the updates (flake8, I think) gave me a code style error about this.

File: machine_common_sense/parameter.py [hunk 3]
```diff
@@ -378,9 +378,9 @@ def _get_object_vector(self, **kwargs) -> Dict:
         )
 
     def _validate_and_convert_params(self, **kwargs) -> Tuple[Action, Dict]:
-        """Need a validation/conversion step for what ai2thor will accept as input
-        to keep parameters more simple for the user (in this case, wrapping
-        rotation degrees into an object)
+        """Need a validation/conversion step for what ai2thor will accept as
+        input to keep parameters more simple for the user (in this case,
+        wrapping rotation degrees into an object)
```
Comment (ThomasSchellenbergNextCentury) [inline machine_common_sense/parameter.py:383]: Newer versions of flake8 apparently check line lengths for docstring comments now.

**Comment by jaudick** [comment]
@ThomasSchellenbergNextCentury I am getting a failure on integration tests using 3.10, here is the process I used:

1. git clone https://github.com/NextCenturyCorporation/MCS.git
2. cd MCS    
3. python3.10 -m venv --prompt mcs venv  
4. source venv/bin/activate    
5. python -m pip install --upgrade pip setuptools wheel
6. python -m pip install machine-common-sense
7. pip install -r requirements.txt
8. python -m unittest  (Good)

_**- python run_tests.py  --mcs_unity_build_file_path ../../../unity/test19.app/Contents/MacOS/MCS-AI2-THOR --dev (Bad)**_

========================================
HANDMADE TEST METADATA TIER: LEVEL1
Initialize return: {'cameraNearPlane': 9.999999747378752e-05, 'cameraFarPlane': 9999.0}
RUNNING SCENE: 001.empty_room_movement.scene.json
Traceback (most recent call last):
  File "/Users/jacob.audick_cn/Documents/new-requirements/MCS/integration_tests/run_tests.py", line 21, in <module>
    start_handmade_tests(
  File "/Users/jacob.audick_cn/Documents/new-requirements/MCS/integration_tests/run_handmade_tests.py", line 355, in start_handmade_tests
    successful, status = run_single_scene(
  File "/Users/jacob.audick_cn/Documents/new-requirements/MCS/integration_tests/run_handmade_tests.py", line 226, in run_single_scene
    step_metadata = controller.start_scene(scene_data)
  File "/Users/jacob.audick_cn/Documents/new-requirements/MCS/venv/lib/python3.10/site-packages/machine_common_sense/controller.py", line 365, in start_scene
    scene_config = self._convert_scene_config(config_data)
  File "/Users/jacob.audick_cn/Documents/new-requirements/MCS/venv/lib/python3.10/site-packages/machine_common_sense/controller.py", line 345, in _convert_scene_config
    return schema.load(config_data)
  File "/Users/jacob.audick_cn/Documents/new-requirements/MCS/venv/lib/python3.10/site-packages/marshmallow/schema.py", line 722, in load
    return self._do_load(
  File "/Users/jacob.audick_cn/Documents/new-requirements/MCS/venv/lib/python3.10/site-packages/marshmallow/schema.py", line 909, in _do_load
    raise exc
marshmallow.exceptions.ValidationError: {'_schema': ['Invalid input type.']}

**Comment by jaudick** [comment]
@ThomasSchellenbergNextCentury I am getting a failure on integration tests using 3.10. Looks lie is marshmallow. Here is the process I used:

1. git clone https://github.com/NextCenturyCorporation/MCS.git
2. cd MCS    
3. git checkout MCS-1501-reqs
4. python3.10 -m venv --prompt mcs venv  
5. source venv/bin/activate    
6. python -m pip install --upgrade pip setuptools wheel
7. python -m pip install machine-common-sense
8. pip install -r requirements.txt
9. python -m unittest  (Good)

_**- python run_tests.py  --mcs_unity_build_file_path ../../../unity/test19.app/Contents/MacOS/MCS-AI2-THOR --dev (Bad)**_

========================================
HANDMADE TEST METADATA TIER: LEVEL1
Initialize return: {'cameraNearPlane': 9.999999747378752e-05, 'cameraFarPlane': 9999.0}
RUNNING SCENE: 001.empty_room_movement.scene.json
Traceback (most recent call last):
  File "/Users/jacob.audick_cn/Documents/new-requirements/MCS/integration_tests/run_tests.py", line 21, in <module>
    start_handmade_tests(
  File "/Users/jacob.audick_cn/Documents/new-requirements/MCS/integration_tests/run_handmade_tests.py", line 355, in start_handmade_tests
    successful, status = run_single_scene(
  File "/Users/jacob.audick_cn/Documents/new-requirements/MCS/integration_tests/run_handmade_tests.py", line 226, in run_single_scene
    step_metadata = controller.start_scene(scene_data)
  File "/Users/jacob.audick_cn/Documents/new-requirements/MCS/venv/lib/python3.10/site-packages/machine_common_sense/controller.py", line 365, in start_scene
    scene_config = self._convert_scene_config(config_data)
  File "/Users/jacob.audick_cn/Documents/new-requirements/MCS/venv/lib/python3.10/site-packages/machine_common_sense/controller.py", line 345, in _convert_scene_config
    return schema.load(config_data)
  File "/Users/jacob.audick_cn/Documents/new-requirements/MCS/venv/lib/python3.10/site-packages/marshmallow/schema.py", line 722, in load
    return self._do_load(
  File "/Users/jacob.audick_cn/Documents/new-requirements/MCS/venv/lib/python3.10/site-packages/marshmallow/schema.py", line 909, in _do_load
    raise exc
marshmallow.exceptions.ValidationError: {'_schema': ['Invalid input type.']}

**Comment by ThomasSchellenbergNextCentury** [comment]
@jaudick Oof, I forgot to check the integration tests for other python versions... Let me try to figure out if I can install Python 3.10, so I can test this locally

**Comment by ThomasSchellenbergNextCentury** [comment]
@jaudick There were two issues:
1. I forgot to update the versions in the `setup.py` file, and make sure it supports 3.10 (apparently I misread it, and we only supported up to 3.9 before, but now we fully support 3.10 as well!). Please pull the latest.
2. Your step number 7 is incorrect. That command installs the latest release of `machine_common_sense`, ignoring the modified code on your local branch. You need to run: `python -m pip install -e .` (remember the dot at the end). You can verify this refers to your local branch by running `pip show machine_common_sense`; the "Location" should be the file path to your local MCS repo.

The integration tests are currently running fine for me with 3.10, so hopefully this will fix your issue as well.

**Review by jaudick — APPROVED** [review]
Tried the new changes, it works now

**Review by rartiss55 — APPROVED** [review]
Upgrading to Python 3.10 and using this branch seemed to resolve issues I was having with installing requirements.txt on my recently acquired Mac using Python 3.9, so thanks!

**Review by brianpippin — APPROVED** [review]

**Review by phile-caci — APPROVED** [review]
pip installed the new requirements.txt
ran unittests, all pass
looks good

**Pull request merged**